src/components/Spotlight/Spotlight.tsx

```diff
@@ -275,32 +285,42 @@ const Spotlight = ({
   ]);
 
   // TODO: use a resize observer to detect when the bounds of the target change
-  const updateWindowBounds = () => {
+  const updateWindowBounds = useCallback(() => {
     setWindowDimensions({
       width: window.innerWidth,
       height: window.innerHeight,
     });
-  };
+  }, [setWindowDimensions]);
 
-  const updateScrollPosition = (e: Event) => {
+  const updateScrollPosition = useCallback(() => {
     // eslint-disable-next-line @typescript-eslint/ban-ts-comment
     // @ts-ignore documentElement does exist on target element
-    setScrollTop(e?.target?.documentElement.scrollTop);
-  };
+    // setScrollTop(e?.target?.documentElement.scrollTop);
+    setScrollTop(window.scrollY);
+  }, [setScrollTop]);
+
+  useEffect(() => {
+    updateScrollPosition(); // update scroll position every render that the scroll changes so the previous value will update after scrolling events stop
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [window.scrollY]);
+  useEffect(() => {
+    updateWindowBounds(); // update window dimensions every render that the window resizes so the previous value will update after resize events stop
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [window.innerWidth, window.innerHeight]);
 
   useEffect(() => {
     window.addEventListener('scroll', updateScrollPosition);
 
     return () => window.removeEventListener('scroll', updateScrollPosition);
-  }, []);
+  }, [updateScrollPosition]);
 
   useEffect(() => {
     window.addEventListener('resize', updateWindowBounds);
 
     return () => {
       window.removeEventListener('resize', updateWindowBounds);
     };
-  }, []);
+  }, [updateWindowBounds]);
```

Comment:
Could you use something like the top answer here to track the window size? 
https://stackoverflow.com/questions/62846043/react-js-useeffect-with-window-resize-event-listener-not-working

Reply:
Yeah - I think I was just trying to fix what was already there, but a refactor is probably what is really needed.